cpp/lib/src/MarcUtil.cc

```diff
@@ -525,12 +530,64 @@ size_t Record::findFieldsInLocalBlock(const std::string &field_tag, const std::s
 }
 
 
-void Record::write(FILE * const output) const {
+void Record::write(File * const output) const {
     const std::string &raw_record(getRawRecord());
 
-    const size_t write_count(std::fwrite(raw_record.data(), 1, raw_record.size(), output));
+    const size_t write_count(output->write(raw_record.data(), raw_record.size()));
     if (write_count != raw_record.size())
-        Error("Failed to write " + std::to_string(raw_record.size()) + " bytes to MARC output!");
+	throw std::runtime_error("Failed to write " + std::to_string(raw_record.size()) + " bytes to MARC output!");
+}
+
+
+void Record::write(XmlWriter * const xml_writer) const {
+    xml_writer->openTag("record");
+
+    xml_writer->openTag("leader", /* suppress_newline = */ true);
+    (*xml_writer->getFile()) << leader_.toString();
```

Comment:
Wouldn't it be nice to have a xml_writer->write(String) or xml_write->writeTagContent(String)?